.github/workflows/integration-tests.yml

```diff
@@ -102,24 +102,89 @@ jobs:
 
   psycopg2-check:
     name: "Test psycopg2 build version"
-    runs-on: ${{ matrix.scenario.platform }}
+    runs-on: ${{ matrix.platform }}
     strategy:
       fail-fast: false
       matrix:
-        scenario:
-          - {platform: ubuntu-latest, psycopg2-name: psycopg2}
-          - {platform: macos-12, psycopg2-name: psycopg2-binary}
+        platform: [ubuntu-22.04, macos-12]
+        python-version: ["3.8", "3.11"]
     steps:
       - name: "Check out repository"
         uses: actions/checkout@v4
 
-      - name: "Test psycopg2 name"
+      - name: "Set up Python ${{ matrix.python-version }}"
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: "Test psycopg2 name - default"
+        run: |
+          python -m venv venv
+          source venv/bin/activate
+          python -m pip install .
+
+          PSYCOPG2_NAME=$((pip show psycopg2 || pip show psycopg2-binary) | grep Name | cut -d " " -f 2)
+          if [[ "$PSYCOPG2_NAME" != "$PSYCOPG2_EXPECTED_NAME" ]]; then
+            exit 1
+          fi
+
+          deactivate
+          rm -r ./venv
+        env:
+          PSYCOPG2_EXPECTED_NAME: psycopg2-binary
+
+      - name: "Test psycopg2 name - invalid override"
         run: |
+          python -m venv venv
+          source venv/bin/activate
           python -m pip install .
-          PSYCOPG2_PIP_ENTRY=$(pip list | grep "psycopg2 " || pip list | grep psycopg2-binary)
-          echo $PSYCOPG2_PIP_ENTRY
-          PSYCOPG2_NAME="${PSYCOPG2_PIP_ENTRY%% *}"
-          echo $PSYCOPG2_NAME
-          if [[ "${PSYCOPG2_NAME}" != "${{ matrix.scenario.psycopg2-name }}" ]]; then
+
+          PSYCOPG2_NAME=$((pip show psycopg2 || pip show psycopg2-binary) | grep Name | cut -d " " -f 2)
+          if [[ "$PSYCOPG2_NAME" != "$PSYCOPG2_EXPECTED_NAME" ]]; then
             exit 1
           fi
+
+          deactivate
+          rm -r ./venv
+        env:
+          DBT_PSYCOPG2_NAME: rubber-baby-buggy-bumpers
+          PSYCOPG2_EXPECTED_NAME: psycopg2-binary
+
+      - name: "Test psycopg2 name - override"
+        run: |
+          python -m venv venv
+          source venv/bin/activate
+          python -m pip install .
+
+          PSYCOPG2_NAME=$((pip show psycopg2 || pip show psycopg2-binary) | grep Name | cut -d " " -f 2)
+          if [[ "$PSYCOPG2_NAME" != "$PSYCOPG2_EXPECTED_NAME" ]]; then
+            exit 1
+          fi
+
+          deactivate
+          rm -r ./venv
+        env:
+          DBT_PSYCOPG2_NAME: psycopg2
+          PSYCOPG2_EXPECTED_NAME: psycopg2-binary  # we have not implemented the hook yet, so this doesn't work
+
+      - name: "Test psycopg2 name - manual override"
+        run: |
+          python -m venv venv
+          source venv/bin/activate
+          python -m pip install .
+
+          # this is the work around
```

Comment:
can you explain why it's needed?

Reply:
Why the workaround is needed? Or why the test is needed?

The workaround is needed because I couldn't get the automation to work.

The test is needed in case we ever do something that makes the workaround, documented in the `README.md`, not work. So it's a way of making sure we update the docs if we ever break it.